### PR TITLE
SNOW-984877 Update to support customized url and add customer name in request header

### PR DESCRIPTION
**Pull request by sfc-gh-dwang**
SNOW-984877  idea
Update to support customized url and option to add account name in request http header

### CODE REVIEW

**Comment by sfc-gh-dwang** [comment]
@sfc-gh-tzhang @sfc-gh-tjones can you take a look if this change is ok?
basically we have 2 changes needed to support localhost:
1. SnowhouseURL to support localhost as url, no account name in the url path
2. we need to add the snowflake-account header in the requestbuilder
I tested in local devvm, and it works, but likely need to test in temptest env
Can we deploy a snapshot unshaded jar?

File: src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java [hunk 1]
```diff
@@ -151,13 +148,36 @@ public class SnowflakeStreamingIngestClientInternal<T> implements SnowflakeStrea
...
-      boolean isTestMode,
-      RequestBuilder requestBuilder,
-      Map<String, Object> parameterOverrides) {
+          String name,
+          SnowflakeURL accountURL,
+          Properties prop,
+          CloseableHttpClient httpClient,
+          boolean isTestMode,
+          RequestBuilder requestBuilder,
+          Map<String, Object> parameterOverrides) {
```
Comment (sfc-gh-tjones) [inline src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java:157]: Formatting looks wrong

**Review by sfc-gh-tjones — APPROVED** [review]

File: src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java (hunk 1 above)
Comment (sfc-gh-dwang) [reply src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java]: updated

**Comment by sfc-gh-dwang** [comment]
@sfc-gh-tjones can you take another check?

File: src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java [hunk 2]
```diff
@@ -59,10 +59,7 @@
 import javax.management.ObjectName;
 import net.snowflake.client.core.SFSessionProperty;
 import net.snowflake.client.jdbc.internal.apache.http.impl.client.CloseableHttpClient;
-import net.snowflake.ingest.connection.IngestResponseException;
-import net.snowflake.ingest.connection.OAuthCredential;
-import net.snowflake.ingest.connection.RequestBuilder;
-import net.snowflake.ingest.connection.TelemetryService;
+import net.snowflake.ingest.connection.*;
```
Comment (sfc-gh-tjones) [inline src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java:62]: Nit, no wildcard imports.

**Review by sfc-gh-tjones — APPROVED** [review]

File: src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java (hunk 2 above)
Comment (sfc-gh-dwang) [reply src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java:62]: did not notice this was changed to wildcard, will fix, thanks for the catching

**Pull request merged**